bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/api/CatalogQuery.scala

```diff
@@ -41,30 +49,87 @@ sealed trait CatalogQuery {
 
+trait CatalogQueryWithRange extends CatalogQuery { this: CatalogQuery =>
+  val magnitudeRange: MagnitudeRange
+}
+
 object CatalogQuery {
-  private case class BandConstrainedCatalogQuery(base: Coordinates, radiusConstraint: RadiusConstraint, magnitudeConstraints: Option[MagnitudeConstraints], catalog: CatalogName) extends CatalogQuery {
-    override val magnitudeRange: Option[MagnitudeRange] = None
-    override def withMagnitudeConstraints(magnitudeConstraints: Option[MagnitudeConstraints]): CatalogQuery = copy(magnitudeConstraints = magnitudeConstraints)
+  private case class BandConstrainedCatalogQuery(base: Coordinates, radiusConstraint: RadiusConstraint, magnitudeConstraints: MagnitudeConstraints, catalog: CatalogName) extends CatalogQuery {
+    override val filters: List[QueryResultsFilter] = List(RadiusFilter(base, radiusConstraint), MagnitudeFilter(magnitudeConstraints))
   }
 
-  private case class RangeConstrainedCatalogQuery(base: Coordinates, radiusConstraint: RadiusConstraint, magnitudeRange: Option[MagnitudeRange], catalog: CatalogName) extends CatalogQuery {
-    override val magnitudeConstraints = None
+  private case class RangeConstrainedCatalogQuery(base: Coordinates, radiusConstraint: RadiusConstraint, catalog: CatalogName) extends CatalogQuery {
+    override val filters: List[QueryResultsFilter] = List(RadiusFilter(base, radiusConstraint))
   }
 
-  private case class GemsCatalogQuery(override val id: Option[Int], base: Coordinates, radiusConstraint: RadiusConstraint, magnitudeRange: Option[MagnitudeRange], catalog: CatalogName) extends CatalogQuery {
-    override val magnitudeConstraints: Option[MagnitudeConstraints] = None
-    override def withMagnitudeRange(magnitudeRange: Option[MagnitudeRange]): CatalogQuery = copy(magnitudeRange = magnitudeRange)
+  private case class RangeConstrainedCatalogQueryWithBand(base: Coordinates, radiusConstraint: RadiusConstraint, mag: MagnitudeExtractor, magnitudeRange: MagnitudeRange, catalog: CatalogName) extends CatalogQuery with CatalogQueryWithRange {
+    override val filters: List[QueryResultsFilter] = List(RadiusFilter(base, radiusConstraint), MagnitudeValueFilter(mag, magnitudeRange))
   }
 
-  // TODO Ensure that it makes sense to default to ucac4
-  def catalogQuery(base: Coordinates, radiusConstraint: RadiusConstraint, magnitudeConstraints: Option[MagnitudeConstraints], catalog: CatalogName = ucac4): CatalogQuery
-    = BandConstrainedCatalogQuery(base, radiusConstraint, magnitudeConstraints, catalog)
+  private case class CatalogQueryWithAdjustedBand(override val id: Option[Int], base: Coordinates, radiusConstraint: RadiusConstraint, mag: MagnitudeExtractor, magnitudeRange: MagnitudeRange, rangeAdjustment: MagnitudeRangeAdjustment, catalog: CatalogName) extends CatalogQuery with CatalogQueryWithRange {
+    override val filters: List[QueryResultsFilter] = List(RadiusFilter(base, radiusConstraint), AdjustedMagnitudeValueFilter(mag, magnitudeRange, rangeAdjustment))
+  }
```

Comment:
That is good point, the multiple classes had different behaviors but now that they are unified they could likely be removed and make `CatalogQuery` a concrete class